### PR TITLE
[arser] Make arser print short option properly

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit makes arser print short option properly.

Even though #6535,  #6572 have been merged, arser didn't print the short option properly because they were just about an interface-related changes. From this PR, I expect arser will be able to fully support a short option.

I think it is a bit hard to review the changes but I will try to explain it as much as I can.

Here are changes that I think is important.
- `Argument` class has introduced `_short_name`, `_long_name`, `_names`.
  - `_name` has been divided into `_short_name` and `_long_name` because `arser` got to support multiple names.
  - `_names` vector includes `_short_name` and `_long_name`. This variable is for pretty printing.
- Validation of arguments given by users.
  - Multiple names requires more thorough validation than it used to.
- Comments for developers.

Related: #4433 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: compiler/arser/include/arser/arser.h
Comment (seanshpark) [inline compiler/arser/include/arser/arser.h:76]: Please use one style `/**` or `///`

File: compiler/arser/include/arser/arser.h
Comment (seanshpark) [inline compiler/arser/include/arser/arser.h:594]: this seems not related with the change

File: compiler/arser/include/arser/arser.h
Comment (mhs4670go) [reply compiler/arser/include/arser/arser.h:76]: @seanshpark Oh, I thought `///` is for single line and `/**` for larger line. Should it be fixed?

Comment (seanshpark) [reply compiler/arser/include/arser/arser.h:76]: > Oh, I thought /// is for single line and /** for larger line.

Actually we didn't have any strict guide for this but usually we used `/**` style, even for one line.

> Should it be fixed?

I hope so...

File: compiler/arser/include/arser/arser.h
Comment (seanshpark) [inline compiler/arser/include/arser/arser.h:153]: (Q) does `---` or `----` work?

File: compiler/arser/include/arser/arser.h
Comment (seanshpark) [inline compiler/arser/include/arser/arser.h:154]: Q) is `-hello` valid?

File: compiler/arser/include/arser/arser.h
Comment (seanshpark) [inline compiler/arser/include/arser/arser.h:161]: What I assumed is, long option with full name, like `--help`and short option with one dash and one character like `-h`
from this comment, I has some questions like above...

**Comment by seanshpark** [comment]
Code itself is a bit hard to understand... I only can do what I can...
Spliting the change will help bit it's an option.

File: compiler/arser/include/arser/arser.h
Comment (mhs4670go) [reply compiler/arser/include/arser/arser.h:153]: @seanshpark Yes. It works well. Which one will be long or short is intentionally not be described because that is so verbose.

File: compiler/arser/include/arser/arser.h
Comment (mhs4670go) [reply compiler/arser/include/arser/arser.h:154]: ditto

File: compiler/arser/include/arser/arser.h
Comment (mhs4670go) [reply compiler/arser/include/arser/arser.h:161]: ditto. That's the basic assumption. Even though user gives `----h` or etc, it will work.

**Comment by mhs4670go** [comment]
@seanshpark 
> Code itself is a bit hard to understand... I only can do what I can...
Spliting the change will help bit it's an option.

Well, I guess the reason why it is hard to understand is because the place to review is a part of printing. For pretty printing, it needs to be complicated. What I come up with is to merge this PR after #6592 is merged. 
> Add help message test that outputs of `./driver -h`.

At least, you will be able to see the print message, which will be basis for reviewing this PR.

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
> At least, you will be able to see the print message, which will be basis for reviewing this PR.

I think we can go with this :)

**Pull request merged**
